lib/ansible/modules/network/ios/ios_ntp.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+DOCUMENTATION = '''
+---
+module: ios_ntp
+extends_documentation_fragment: ios
+version_added: "2.8"
+short_description: Manages core NTP configuration.
+description:
+    - Manages core NTP configuration.
+author:
+    - Federico Olivieri (@Federico87)
+options:
+    server:
+        description:
+            - Network address of NTP server.
+    source_int:
+        description:
+            - Source interface for NTP packets.
+    acl:
+        description:
+            - ACL for peer/server access restricition.
+    logging:
+        description:
+            - Enable NTP logs. Data type boolean.
+        type: bool
+        default: False
+        choices: [True, False]
+    auth:
+        description:
+            - Enable NTP authentication. Data type boolean.
+        type: bool
+        default: False
+        choices: [True, False]
+    auth_key:
+        description:
+            - md5 NTP authentication key of tye 7.
+    key_id:
+        description:
+            - auth_key id. Datat type string
+    state:
+        description:
+            - Manage the state of the resource.
+        default: present
+        choices: ['present', 'absent']
+'''
+
+EXAMPLES = '''
+# Set NTP Server with parameters
+- ios_ntp:
+    server: 10.0.255.10
+    source_int: Loopback0
+    acl: NTP_ACL
+    logging: true
+    key_id: 10
+    auth_key: 15435A030726242723273C21181319000A
+    auth: true
+    state: present
+    provider: "{{ staging }}"
```

Comment:
`provider` dict has been deprecated. Starting with Ansible 2.5 we recommend using `connection: network_cli`.
New modules should refrain from using it in examples, IMO.

Reply:
Removed.